lib/deimos/utils/schema_class.rb

```diff
@@ -10,10 +10,23 @@ class << self
         # @return [Array<String>]
         def modules_for(namespace)
           modules = ['Schemas']
-          namespace_folder = namespace.split('.').last
-          if Deimos.config.schema.generate_namespace_folders && namespace_folder
-            modules.push(namespace_folder.underscore.classify)
+
+          if Deimos.config.schema.generate_namespace_folders
+            namespace_map = Deimos.config.schema.schema_namespace_map.sort_by{|k,v| -k.length}.to_h
+            namespace_override = namespace_map.keys.detect{ |k| namespace.include?(k) }
+
+            module_namespace = if namespace_override.present?
+              # don't use Schemas default module
```

Comment:
This also should be called out in documentation. Is there a reason we don't want to use the default module as the base?

Reply:
Will update the docs. The reason to not use the default is you may want to use a different module name so it makes it more configurable.

For example, we have the Avro schemas in a schemas directory and the generated classes in a schema_classes directory so we want our namespace map to look like this:
```
schema_namespace_map({
        'com.org.suborg' => 'SchemaClasses',
        'com.org.suborg.avro' => 'SchemaClasses', # no need to include an avro module
        'com.org.suborg.avro.somethingelse' => 'SchemaClasses' # too much nesting otherwise
      })
```
For our use case, we really just need a configurable array of namespace prefixes to squash and the ability to set the default module, but having the map makes it highly configurable for other use cases.